quiz/quizfinal.js

```diff
@@ -77,128 +70,207 @@ function getRandomQuestions(quizQuestionsArr) {
 
   let quantity = 0;
   while (quantity < quizQuestionsQuantity) {
-    // for (let i = 0; i < quizQuestionsArr.length; i++) {
     const randomQuestionIndex = Math.floor(
       Math.random() * quizQuestionsArr.length
     );
     randomQuiz.push(quizQuestionsArr[randomQuestionIndex]);
     quizQuestionsArr.splice(randomQuestionIndex, 1);
-    // console.log("randomQuiz", randomQuiz);
     quantity++;
-
-    // }
   }
   randomQuiz.sort((a, b) => a.level - b.level);
-  // console.log("randomQuiz", randomQuiz);
 
   return randomQuiz;
 }
 
-// console.log("randomQuiz", randomQuiz);
+let formElem = document.querySelector("form");
+formElem.addEventListener("submit", (event) => {
+  event.preventDefault();
+});
 
-console.log(getRandomQuestions(quizQuestions.math));
+function renderAnswerOptions(quiz, currentIndex, isResult) {
+  // console.log("currentIndex", currentIndex);
+  // console.log("randomQuiz", randomQuiz);
 
-const nextBtn = document.querySelector("#next-btn");
+  const answersArray = quiz[currentIndex]["answersArray"];
+  const answersList = document.createElement("ul");
+  answersList.classList.add("answers");
 
-let formElem = document.querySelector("form");
+  for (let answerObject of answersArray) {
+    const answerValue = answerObject.answer;
+    const answerLine = document.createElement("li");
+    answerLine.classList.add("answer");
+    answerLine.id = "answer";
+    const radioButton = document.createElement("input");
+    radioButton.setAttribute("type", "radio");
+    radioButton.name = "answer";
+    radioButton.id = "answer";
+    radioButton.value = answerValue;
+
+    const label = document.createElement("label");
+    label.append(radioButton);
+    answerLine.append(label);
+    answerLine.append(answerValue);
+
+    const currentQuestion = quiz[currentIndex];
```

Comment:
I'd move this inside the if statement below since it's not used anywhere else. You want to declare variables as close to their usage as possibel.